chain/beacon/drand/drand.go

```diff
@@ -113,6 +112,18 @@ func NewDrandBeacon(genesisTs, interval uint64, ps *pubsub.PubSub, config dtypes
 		opts = append(opts, gclient.WithPubsub(ps))
 	} else {
 		log.Info("drand beacon without pubsub")
+		if len(clients) == 0 {
+			// This hack is necessary to convince a drand beacon to start without any clients. For
+			// historical becaons we need them to be able to verify old entries but we don't need to fetch
+			// new ones. With pubsub enabled, it acts as a client so drand is happy, but if we don't have
+			// pubsub then drand will complain about old beacons withotu clients. So we make one that
+			// it'll think is a valid client and that it won't speed test (hence the need to mark it as
+			// as "watcher").
+			historicalClient := &historicalBeaconClient{}
+			opts = append(opts, dclient.WithWatcher(func(chainInfo *dchain.Info, cache dclient.Cache) (dclient.Watcher, error) {
+				return historicalClient, nil
```

Comment:
I think the trick is to have your client's `Get` return a `drand.ErrEmptyClientUnsupportedGet` error to avoid having to have a "valid" client in `Wrap`.

Reply:
Looking into this, it's possible we're not properly "supporting" that special error in the `testSpeed` function.

I could change it if it'd help, so that we're not reporting the 
``` 
oc.log.Infow("", "optimizing_client", "endpoint down when speed tested", ...
``` 
log line when it sees that error.

Reply:
yep, leave it long enough and we get that:

```
2025-02-04T16:29:52.276+1100    INFO    drand   client/optimizing.go:194                {"optimizing_client": "endpoint down when speed tested", "client": "EmptyClient.(+verifier)", "err": "not supported"}
2025-02-04T16:29:52.276+1100    INFO    drand   client/optimizing.go:194                {"optimizing_client": "endpoint down when speed tested", "client": "EmptyClient.(+verifier)", "err": "not supported"}
```

Opened https://github.com/drand/go-clients/issues/10 for this